Platform/ElkhartlakeBoardPkg/Script/StitchIfwiConfig.py

```diff
@@ -74,49 +74,29 @@ def get_platform_sku():
-      ('IFWI/BIOS/TS1/ACM0',      'Input/acm.bin',        'dummy',    '',                            ''),
-    ]
+      # Path                      file name                  compress  Key                              SVN
+      ('IFWI/BIOS/TS0/ACM0',      'Input/acm.bin',           'dummy',  '',                              ''),
+      ('IFWI/BIOS/TS1/ACM0',      'Input/acm.bin',           'dummy',  '',                              ''),
 
-    if os.path.exists('IPFW/PreOsChecker.bin'):
-        print ("PreOsChecker.bin found")
-        replace_list.append (
-            ('IFWI/BIOS/NRD/IPFW/POSC', 'IPFW/PreOsChecker.bin', 'dummy', 'KEY_ID_CONTAINER_COMP_RSA3072', 0), # Pre-OS Checker
-        )
-    if os.path.exists('IPFW/PseFw.bin'):
-        print ("PseFw.bin found")
-        replace_list.append (
-            ('IFWI/BIOS/NRD/IPFW/PSEF', 'IPFW/PseFw.bin',          'lz4',     'KEY_ID_CONTAINER_COMP_RSA3072', 0),   # OSE FW
-        )
-    if 'tsn' in plt_params_list:
-        if os.path.exists('IPFW/PseTsnIpConfig.bin'):
-            print ("PseTsnIpConfig.bin found")
-            replace_list.append (
-                ('IFWI/BIOS/NRD/IPFW/TSIP', 'IPFW/PseTsnIpConfig.bin', 'lz4',     'KEY_ID_CONTAINER_COMP_RSA3072', 0),   # PSE TSN IP
-            )
-        if os.path.exists('IPFW/TsnConfig.bin'):
-            print ("TsnConfig.bin found")
-            replace_list.append (
-                ('IFWI/BIOS/NRD/IPFW/TSNC', 'IPFW/TsnConfig.bin',      'lz4',     'KEY_ID_CONTAINER_COMP_RSA3072', 0),   # TSN Config
-            )
-        if os.path.exists('IPFW/TsnMacAddr.bin'):
-            print ("TsnMacAddr.bin found")
-            replace_list.append (
-                ('IFWI/BIOS/NRD/IPFW/TMAC', 'IPFW/TsnMacAddr.bin',     'lz4',     'KEY_ID_CONTAINER_COMP_RSA3072', 0),   # TSN MAC Address
-            )
-    if os.path.exists('IPFW/crl.bin'):
-        print ("crl.bin found")
-        replace_list.append (
-            ('IFWI/BIOS/NRD/IPFW/TCCM', 'IPFW/crl.bin',         'lz4',     'KEY_ID_CONTAINER_COMP_RSA3072', 0),   # TCC CRL binary
-        )
+      # need to set ENABLE_PRE_OS_CHECKER = 1 in BoardConfig.py
+      ('IFWI/BIOS/NRD/IPFW/POSC', 'IPFW/PreOsChecker.bin',   'dummy',  'KEY_ID_CONTAINER_COMP_RSA3072', 0 ), # Pre-OS Checker
+
+      # need to set ENABLE_PSEFW_LOADING = 1 in BoardConfig.py
+      ('IFWI/BIOS/NRD/IPFW/PSEF', 'IPFW/PseFw.bin',          'lz4',    'KEY_ID_CONTAINER_COMP_RSA3072', 0 ), # OSE FW
+
+      # need to set ENABLE_TSN = 1 in BoardConfig.py
+      # ('IFWI/BIOS/NRD/IPFW/TSIP', 'IPFW/PseTsnIpConfig.bin', 'lz4',    'KEY_ID_CONTAINER_COMP_RSA3072', 0 ), # PSE TSN IP
+      # ('IFWI/BIOS/NRD/IPFW/TSNC', 'IPFW/TsnConfig.bin',      'lz4',    'KEY_ID_CONTAINER_COMP_RSA3072', 0 ), # TSN Config
+      # ('IFWI/BIOS/NRD/IPFW/TMAC', 'IPFW/TsnMacAddr.bin',     'lz4',    'KEY_ID_CONTAINER_COMP_RSA3072', 0 ), # TSN MAC Address
```

Comment:
Any reason to remove the file exist check and comment these lines out?
SBL supports to build these binaries into SBL image during build time and stitch time without changing stitch script.  I think we could update StitchIfwi tool if you see an issue (e.g. give a warning if the component is not found in SBL image while replacing component)

Reply:
Take TCC as an example,
if TCC is enabled during build time, but the component is not found in SBL image, then user can still replace it during stitch time.
If TCC is not enabled during build time, but the component is put in the specified IPFW path, then stitch process will fail.
To avoid this case, I would rather that user can explicitly modify the replace_list and be aware of the ENABLE_TCC flag being set, and then put the binary in the specified path.
Alternative way is to use stitch -o options to control the replace_list, so no stitch script change would happen.

Reply:
How about we enhance StitchIfwi.py to give a warning for the case "If TCC is not enabled during build time, but the component is put in the specified IPFW path, then stitch process will fail"?  StitchIfwi could know if a component exists or not in SBL image, it could skip to replace a component if that component doesn't exist.

Reply:
Instead of raise exception, added the code to skip the component replacement if the component file is not existing or if the component is not locatable in SBL image.
Also print the skipped components so user would be aware of.

example output messages:
IFWI Stitching completed successfully !
Boot Guard Profile: FVME
**Skipped replacement components: ['IFWI/BIOS/NRD/IPFW/TSIP', 'IFWI/BIOS/NRD/IPFW/TSNC', 'IFWI/BIOS/NRD/IPFW/TMAC']**
IFWI image: sbl_ifwi_ehl.bin